uwblueprint/planet-read
# Implement two approved languages pipelines

## Notion ticket link



[Implement two approved languages pipelines](https://www.notion.so/uwblueprintexecs/Implement-two-approved-languages-pipelines-6e785865a88a4d06b90f449d98ad48de)



## Implementation description

- changed all instances of `approved_languages` -> `approved_languages_translation`
- generated new migration for users table with columns `approved_languages_translation` and `approved_languages_review`
- changed frontend instances of `approvedLanguages` -> `approvedLanguagesTranslation`
- added `approved_languages_review` to `insert_test_data.py` script



## Steps to test
1. Rebuild containers: `docker-compose down --volumes` and `docker-compose up --build`
2. Verify that the new alembic revision works properly (just check that these commands run properly without failing):
```
$ docker exec -it planet-read_py-backend_1 /bin/bash -c "flask db upgrade"
$ docker exec -it planet-read_py-backend_1 /bin/bash -c "flask db downgrade"
$ docker exec -it planet-read_py-backend_1 /bin/bash -c "flask db upgrade"
```
3. Insert test data into the database: `docker exec -it planet-read_py-backend_1 /bin/bash -c "python -m tools.insert_test_data"`
4. Verify that a new `approved_languages_review` column is added to the users table and `approved_languages` column is renamed: `docker exec -it planet-read_db_1 mysql -u root -proot -e "USE planet-read; SELECT * FROM users;"`
5. Poke around in the frontend and verify that you are still able to login, view stories to translate, and view stories to review
6. Try to translate a story; verify that this opens up the translation platform
7. Try to review a story; this should error out with a `Error: User can't be assigned as a reviewer` alert



## What should reviewers focus on?

- `backend/python/tools/insert_test_data.py`
- For the `insert_test_data.py` script, the `approved_languages_review` were set to be the previous users' `approved_languages_translation` so that each translation can have a corresponding reviewer; however, is this too unrealistic as the users have completely different approved translation/review languages?
- Selecting a story to review in the frontend does not work as it has not been updated to read from `approvedLanguagesReview`

## Checklist
